amd/src/preferencelib.js

```diff
@@ -0,0 +1,74 @@
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Helper for handling user preferences.
+ *
+ * @module     tiny_elements/preferencelib
+ * @copyright  2024 ISB Bayern
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+import Ajax from 'core/ajax';
+import Notification from 'core/notification';
+
+export const Preferences = {
+    category: 'elements_category',
+    // eslint-disable-next-line camelcase
+    category_flavors: 'elements_category_flavors',
+    // eslint-disable-next-line camelcase
+    component_variants: 'elements_component_variants'
+};
+
+/**
+ * Load user preferences.
+ * @param {string} name
+ * @returns {Promise}
+ */
+export const loadPreferences = async(name) => {
```

Comment:
I do not think that is an async function here?

Reply:
Needs to be to await result of AJAX call

Reply:
Made a little bit more clear now